core/remotes/docker/pusher.go

```diff
@@ -176,7 +177,7 @@ func (p dockerPusher) push(ctx context.Context, desc ocispec.Descriptor, ref str
 			//
 			// for the private repo, we should remove mount-from
 			// query and send the request again.
-			resp, err = preq.doWithRetries(pctx, nil)
+			resp, err = preq.doWithRetries(pctx, nil, true)
```

Comment:
should be `false`?  given `host       = hosts[0]`

Reply:
same for the 2 places below

Reply:
Maybe `lastHost` is a bad name. Its more a "try harder" or don't give up quickly. For push or requests after a successful request, we shouldn't give us so quickly though.